Makes legacy Bungee always use the legacy network #166

EpicKnarvik97 opened the pull request:
This change forces all legacy BungeeCord portals to be created on the legacy network.
This change also skips the code trying to make sense of the server name when creating a new legacy BungeeCord portal using a sign.
Random instances of the legacy network name have been changed to get it from the BungeePortal class.
This should fix some of the problems of #166, though the legacy network could use a new name.

EpicKnarvik97 committed:
Makes legacy Bungee always use the legacy network

EpicKnarvik97 committed:
Makes the legacy network name easily changeable

EpicKnarvik97 committed:
Merge branch 'dev' of https://github.com/stargate-rewritten/Stargate-Bukkit into dev

EpicKnarvik97 committed:
Fixes some warnings in GateFormatReader

EpicKnarvik97 committed:
Merge branch 'dev' into bungee-fix

Pheotis reviewed (changes requested):
LGTM! A few notes:

- The `§§§§§§#BUNGEE#§§§§§§` string was originally intended to be something that no user could accidentally end up typing.
  - That said, this seems likely to cause confusion with the networks `(bungee)` and `{bungee}`.
  - I think `LegacyBungee` is probably less likely to cause confusion.
  - This string should probably be configurable via the [AdvancedConfig.yml](https://github.com/stargate-rewritten/Stargate-Bukkit/wiki/Configuration-Guide#advanced_configyml) system.
- This network should probably be reserved in a similar manner to default and terminals.
   - As a reserved network, it should be impossible to create non-legacy-bungee portals on it.
   - If the user "{the legacy bungee string}" joins, their personal network should be forced as `{65e439cc}` (the first 8 characters of their UUID).
   - stargate.*.network.custom.{the bungee string} should be ignored.
   - Instead, a unique check should occur, although I am personally torn between the below:
     - We could add a `stargate.*.network.legacybungee node`.
     - That might be redundant though; if the user has perms for *any* network, and has stargate.*.type.u, they should probably be allowed to use legacy bungee portals.

EpicKnarvik97 committed:
Fixes some BungeeCord-related "problems"

Changes the legacy BungeeCord network to LegacyBungee
Prevents users from creating Stargates on the legacy BungeeCord network
Fixes some potential crashes related to the instantiation of the LegacyBungee network

EpicKnarvik97 commented:
> * We could add a `stargate.*.network.legacybungee node`.

I can require sg.network.legacybungee, require sg.network or perhaps just require nothing for the network permission. It's kind of a really special case as you can only use it for legacy BungeeCord portals, and it usage already requires the user to be able to use BungeeCord portals. It's not really a chosen network, but a consequence of creating a legacy BungeeCord portal.
A user may be able to create BungeeCord portals without having sg.network though.

For now, I've gone the allow route, as requring a separate permission for the legacy bungee network only makes it confusing.

EpicKnarvik97 committed:
Skips legacy BungeeCord network permission check

This change skips the legacy Bungee network check. This means that if you are able to create a Legacy Bungee Stargate, you are able to create it on the legacy bungee network.
In addition, if for some reason, something tries to create a portal without the legacy bungee flag on the legacy bungee network, an invalid permission node will be returned to hopefully prevent the action.
